packages/lesswrong/components/notifications/NotificationTypeSettings.tsx

```diff
@@ -1,92 +1,169 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import { Components, registerComponent } from '../../lib/vulcan-lib';
 import Select from '@material-ui/core/Select';
 import withErrorBoundary from '../common/withErrorBoundary';
-import PropTypes from 'prop-types';
-import { defaultNotificationTypeSettings, NotificationChannelOption } from '../../lib/collections/users/schema';
+import {
+  DayOfWeek,
+  isLegacyNotificationTypeSettings,
+  LegacyNotificationTypeSettings,
+  legacyToNewNotificationTypeSettings,
+  NotificationBatchingFrequency,
+  NotificationChannel,
+  NotificationChannelSettings,
+  NotificationTypeSettings,
+} from "../../lib/collections/users/schema";
 import { getNotificationTypeByUserSetting } from '../../lib/notificationTypes';
 import type { PickedTime } from '../common/BatchTimePicker';
 import { isFriendlyUI } from '../../themes/forumTheme';
+import classNames from 'classnames';
 
 const styles = (theme: ThemeType) => ({
   root: {
-    padding: 8,
+    paddingLeft: 8,
+    paddingRight: 8,
+    display: "flex",
+    flexDirection: "column",
+    gap: "8px"
   },
   label: {
     fontFamily: isFriendlyUI ? theme.palette.fonts.sansSerifStack : undefined,
   },
-  settings: {
-    paddingLeft: 20,
+  channelLabel: {
+    fontSize: 13
   },
+  channelSettings: {
+    paddingLeft: 8,
+    display: "flex",
+    gap: "8px",
+    justifyContent: "space-between",
+    alignItems: "center",
+    fontSize: 13,
+    [theme.breakpoints.down('sm')]: {
+      paddingLeft: 0,
+    }
+  },
+  channelSettingsDetails: {
+    display: "flex",
+    flexWrap: "wrap",
+    alignItems: "center",
+    gap: "4px",
+    ...theme.typography.body2,
+    fontSize: 13,
+    "&>*": {
+      whiteSpace: "nowrap"
+    }
+  },
+  channelSettingsDisabled: {
+    color: theme.palette.grey[600]
+  },
+  toggle: {
+    alignItems: "center",
+    marginRight: 8,
+    [theme.breakpoints.down('sm')]: {
+      marginRight: 0,
+    }
```

Comment:
The toggle button looks a bit far away from the thing it is toggling when at full width, so I added this margin